R-package/tests/testthat/test_registration.R

```diff
@@ -0,0 +1,17 @@
+ON_WINDOWS <- .Platform$OS.type == "windows"
+
+# this test is used to catch silent errors in routine registration,
+# like the one documented in
+# https://github.com/microsoft/LightGBM/issues/4045#issuecomment-812289182
+test_that("lightgbm routine registration worked", {
+    testthat::skip_if_not(ON_WINDOWS)
```

Comment:
this `skip_if_not()` is to work around this error I saw on Linux jobs in CI

```text
* checking tests ...
  Running ‘testthat.R’
   ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
  ══ Failed tests ════════════════════════════════════════════════════════════════
  ── Failure (test_registration.R:8:5): lightgbm routine registration worked ─────
  dll_info[["dynamicLookup"]] is not FALSE
  
  `actual` is NULL
  `expected` is a logical vector (FALSE)
  ── Error (test_registration.R:11:5): lightgbm routine registration worked ──────
  Error: no applicable method for 'getDLLRegisteredRoutines' applied to an object of class "NULL"
  Backtrace:
      █
   1. └─base::getDLLRegisteredRoutines(dll_info[["path"]]) test_registration.R:11:4
  
  [ FAIL 2 | WARN 0 | SKIP 3 | PASS 634 ]
  Error: Test failures
  Execution halted
```

Reply:
ah! The problem with this test isn't actually about operating system. This test will fail with an error like the one above on LightGBM's CMake-based builds and pass on CRAN builds. That's because, by using `install.libs.R` to build `lib_lightgbm` instead of R's built-in machinery, the routine registration stuff doesn't work.

Given that as of this PR the package will now use literal R symbols (e.g. `.Call(LGBM_BoosterGetCurrentIteration_R,...`), I don't think this test is necessary. Any failures in the registration will result in "object not found errors".

I'm going to remove this test, and will provide more explanation in a separate comment for how to make this work with CMake-based builds.